pkg/parse/namespace.go

```diff
@@ -174,6 +177,51 @@ func (p *namespace) setSourceStatusWithRetries(ctx context.Context, newStatus *S
 	return nil
 }
 
+func (p *namespace) setSourceAnnotations(ctx context.Context, commit string) error {
```

Comment:
This still has a GET call. We can go with a single PATCH call as below:

```
func (p *namespace) setSourceAnnotations(ctx context.Context, commit string) error {
	rs := &v1beta1.RepoSync{}
	rs.Namespace = string(p.Scope)
	rs.Name = p.SyncName

	var patchType types.PatchType
	var patch string
	if p.Options.SourceType == configsync.OciSource ||
		(p.Options.SourceType == configsync.HelmSource && strings.HasPrefix(p.Options.SourceRepo, "oci://")) {
		patchType = types.MergePatchType
		patch = fmt.Sprintf(`{"metadata":{"annotations":{"%s":"%s"}}}`,
			metadata.ImageToSyncAnnotationKey,
			fmt.Sprintf("%s@sha256:%s", p.Options.SourceRepo, commit),
		)
	} else {
		patchType = types.JSONPatchType
		patch = fmt.Sprintf(`[{"op":"remove", "path":"/metadata/annotations/%s"}]`,
			// TODO: `/` in the annotation key needs to be encoded with `~1`,
			// e.g. "/metadata/annotations/configsync.gke.io~1image-to-sync"
			metadata.ImageToSyncAnnotationKey)
	}
	return p.Client.Patch(ctx, rs,
		client.RawPatch(patchType, []byte(patch)),
		client.FieldOwner(configsync.FieldManager))
}
```

Reply:
Done, using mergepatchtype for both, for removing the annotation I'm setting the value to null.